### ISSUE TITLE
Support for open-ended ranges

### ISSUE THREAD

**Issue by jdforsythe** [gf3/moment-range#85]
It would be nice to see support for ranges either without a begin or without an end.

My use case is a user has multiple addresses with dates when they are active. I want to see if the ranges overlap, but they can leave the begin/end date blank to mean forever.

Current workaround is to pick the lowest possible date and some ungodly high date if they're blank, but this is not accurate nor ideal.

**Comment by gf3** [comment]
do you have an idea of what the API for something like this might look like?

**Comment by jdforsythe** [comment]
Off the top of my head... maybe just a third parameter to the constructor, allowOpenRanges, which if true would allow begin or end to be null.

The difficulty is more in the implementation. My hacky way of doing it for now - I made two methods, getEarliestMoment() and getLatestMoment() which get the min and max dates possible. If one or the other is null, I set it to the appropriate date and use that for comparison. It was much easier than reworking all the calculation methods, but not the cleanest way, to be sure.
